stdlib/public/stubs/DispatchShims.mm

```diff
@@ -140,3 +166,27 @@ void SWIFT_DISPATCH_NOESCAPE (^block)(long))
 		return applier(data, off, loc, size);
 	});
 }
+
+void
+swift::_swift_dispatch_source_set_event_handler(
+	__swift_shims_dispatch_source_t source,
+	__swift_shims_dispatch_block_t block)
+{
+	dispatch_source_set_event_handler((dispatch_source_t)source, cast(block));
+}
+
+void
+swift::_swift_dispatch_source_set_cancel_handler(
+	__swift_shims_dispatch_source_t source,
+	__swift_shims_dispatch_block_t block)
+{
+	dispatch_source_set_cancel_handler((dispatch_source_t)source, cast(block));
```

Comment:
Any type which is a typedef of `id` cannot be added to `cast()` without an error.